tests/parametric/test_headers_baggage.py

```diff
@@ -0,0 +1,172 @@
+from operator import le
+from requests import head
+from utils.parametric.spec.trace import SAMPLING_PRIORITY_KEY, ORIGIN
+from utils.parametric.spec.trace import span_has_no_parent
+from utils.parametric.headers import make_single_request_and_get_inject_headers
+from utils.parametric.spec.trace import find_only_span
+from utils import features, scenarios, bug, context
+from typing import Any
+import pytest
+from utils.parametric.spec.trace import SAMPLING_PRIORITY_KEY, ORIGIN
+from utils.parametric.headers import make_single_request_and_get_inject_headers
+from utils.parametric.spec.trace import find_only_span
+from utils import missing_feature, context, scenarios, features, bug
+
+parametrize = pytest.mark.parametrize
+
+import logging
+
+logging.basicConfig(level=logging.INFO)  # or logging.WARNING, logging.ERROR to reduce verbosity
+
+
+def disable_baggage() -> Any:
+    env = {
+        "DD_TRACE_PROPAGATION_STYLE": "datadog,tracecontext",
+    }
+    return parametrize("library_env", [env])
+
+
+def only_baggage_enabled() -> Any:
+    env = {
+        "DD_TRACE_PROPAGATION_STYLE": "baggage",
+    }
+    return parametrize("library_env", [env])
+
+
+@features.datadog_headers_propagation
+@scenarios.parametric
+class Test_Headers_Baggage:
+    def test_distributed_headers_baggage_default_D001(self, test_agent, test_library):
+        """Ensure baggage is enabled as a default setting and that it does not interfere with Datadog headers."""
+        with test_library:
+            headers = make_single_request_and_get_inject_headers(
+                test_library,
+                [
+                    ["x-datadog-trace-id", "123456789"],
+                    ["x-datadog-parent-id", "987654321"],
+                    ["x-datadog-sampling-priority", "2"],
+                    ["x-datadog-origin", "synthetics;=web,z"],
+                    ["x-datadog-tags", "_dd.p.dm=-4"],
+                    ["baggage", "foo=bar"],
+                ],
+            )
+
+        span = find_only_span(test_agent.wait_for_num_traces(1))
+        assert span.get("trace_id") == 123456789
+        assert span.get("parent_id") == 987654321
+        assert "baggage" in headers.keys()
+        assert headers["baggage"] == "foo=bar"
+
+    @only_baggage_enabled()
+    def test_distributed_headers_baggage_only_D002(self, test_library):
+        """Ensure that only baggage headers are injected when baggage is the only enabled propagation style."""
+        with test_library:
+            headers = make_single_request_and_get_inject_headers(test_library, [["baggage", "foo=bar"]])
+
+        assert "x-datadog-trace-id" not in headers.keys()
```

Comment:
`x-datadog-trace-id` and `x-datadog-parent-id` are required span context fields in nodejs, so not setting them would create an error. I can add a skip flag for node